.github/workflows/build_base.yaml

```diff
@@ -3,6 +3,12 @@ name: "Build leap"
 on:
   workflow_dispatch:
   workflow_call:
+    inputs:
+      override-build-matrix:
+        description: 'Override build matrix'
```

Comment:
What's driving the need for adding this? You only want to build leap for a single platform when doing the performance harness test?

Reply:
Correct.

Reply:
Not sure it's worth the added complexity in `build_base.yaml` to accommodate that. Are you doing it just to save a dollar? I'd rather take the cost and not have 30+ more lines of yaml here to maintain forever, especially because PH runs aren't regularly done anyways.

If you want to save cost I'd suggest pushing off the complexity in to the `performance_harness_run.yaml` by doing something like trying to fetch an existing built leap.deb & leap-dev.deb, and then only falling back to building leap if they don't exist.

Reply:
yeah, not having to copy pasta around the `[ubuntuxx..]` is a good improvement. I'm a little worried we're going to have to refactor much of this soon again though because while right now there is a 1:1 between build platforms and test platforms that won't always be the case. That is, right now `ubuntu22` means both building and testing on `ubuntu22` but that may not always be the case. For example, for our reproducible builds we may have a separate `repro` platform that is used for builds, but then tested on _both_ `ubuntu20` and `ubuntu22`. Or ARM builds may have a `crossarm` platform for builds that then runs the tests on `ubuntu20arm` or something. I can't quite wrap my head around the semantics of how that is going to work just yet (besides adding more complexity :disappointed:) so what is in this PR is good for now on that front.

Reply:
That's a good point. There is definitely a complexity addition that will come when that occurs.  It could be that the `Platform Matrix` job provides a nice location to do the platform setup though and could be expanded to help with that?  Might be that it breaks out into its own workflow.  I agree that it is hard to say what will work best until some of those decisions are made.  I'll consider this resolved for now per your comment.